File: sigedu-backend-final/src/main/java/com/sigedu/backend/service/DashboardService.java
package com.sigedu.backend.service;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;
import java.util.Map;

public interface DashboardService {

    // Estadísticas generales
    long getTotalEstudiantes();
    long getTotalEstudiantesActivos();
    long getTotalDocentes();
    long getTotalCursos();
    long getTotalMatriculasAnioActual();

    // Estadísticas por grado
    Map<String, Long> getEstudiantesPorGrado();
    Map<String, Long> getEstudiantesPorNivel();

    // Estadísticas académicas
    Map<String, Double> getPromedioNotasPorCurso(Integer idPeriodoAcademico, Integer bimestre);
    Map<String, Long> getAsistenciaPorEstado(Integer idGradoSeccion, LocalDate fecha);

    // Estadísticas financieras
    BigDecimal getTotalIngresosPorPeriodo(Integer idPeriodoAcademico);
    BigDecimal getTotalDeudasPendientes();
    Map<String, BigDecimal> getIngresosPorMes(Integer anio);
    List<Map<String, Object>> getTopDeudores(int limite);

    // Reportes existentes
    List<Map<String, Object>> getReporteNotasEstudiante(Integer idEstudiante, Integer idPeriodoAcademico);
    List<Map<String, Object>> getReportePagosEstudiante(Integer idEstudiante, Integer idPeriodoAcademico);
    List<Map<String, Object>> getReporteAsistenciaGrado(Integer idGradoSeccion, LocalDate fechaInicio, LocalDate fechaFin);

    // Reportes profesionales agregados en el Paso 12
    Map<String, Object> getResumenEjecutivo(Integer idPeriodoAcademico);
    Map<String, Object> getResumenNotasEstudiante(Integer idEstudiante, Integer idPeriodoAcademico);
    Map<String, Object> getResumenAsistenciaEstudiante(Integer idEstudiante, LocalDate fechaInicio, LocalDate fechaFin);
    List<Map<String, Object>> getPagosPendientes(Integer idPeriodoAcademico);
    Map<String, BigDecimal> getResumenPagosPorEstado(Integer idPeriodoAcademico);
    Map<String, Long> getResumenAsistenciaGeneral(LocalDate fechaInicio, LocalDate fechaFin);
}

File: sigedu-frontend-final/src/api/dashboardService.js
import axios from './axios';

export const dashboardService = {
    getEstadisticasGenerales: async () => {
        const response = await axios.get('/dashboard/stats/generales');
        return response.data;
    },
    getEstudiantesPorGrado: async () => {
        const response = await axios.get('/dashboard/stats/estudiantes-por-grado');
        return response.data;
    },
    getEstudiantesPorNivel: async () => {
        const response = await axios.get('/dashboard/stats/estudiantes-por-nivel');
        return response.data;
    },
    getPromedioNotas: async (idPeriodoAcademico, bimestre) => {
        const response = await axios.get(`/dashboard/stats/promedio-notas?idPeriodoAcademico=${idPeriodoAcademico}&bimestre=${bimestre}`);
        return response.data;
    },
    getAsistencia: async (idGradoSeccion, fecha) => {
        const response = await axios.get(`/dashboard/stats/asistencia?idGradoSeccion=${idGradoSeccion}&fecha=${fecha}`);
        return response.data;
    },
    getIngresosPorPeriodo: async (idPeriodoAcademico) => {
        const response = await axios.get(`/dashboard/stats/ingresos/periodo/${idPeriodoAcademico}`);
        return response.data;
    },
    getTotalDeudas: async () => {
        const response = await axios.get('/dashboard/stats/deudas/total');
        return response.data;
    },
    getIngresosPorMes: async (anio) => {
        const response = await axios.get(`/dashboard/stats/ingresos-por-mes/${anio}`);
        return response.data;
    },
    getTopDeudores: async (limite = 10) => {
        const response = await axios.get(`/dashboard/stats/top-deudores?limite=${limite}`);
        return response.data;
    },
    getReporteNotas: async (idEstudiante, idPeriodoAcademico) => {
        const response = await axios.get(`/dashboard/reportes/notas/estudiante/${idEstudiante}?idPeriodoAcademico=${idPeriodoAcademico}`);
        return response.data;
    },
    getReportePagos: async (idEstudiante, idPeriodoAcademico) => {
        const response = await axios.get(`/dashboard/reportes/pagos/estudiante/${idEstudiante}?idPeriodoAcademico=${idPeriodoAcademico}`);
        return response.data;
    },
    getReporteAsistencia: async (idGradoSeccion, fechaInicio, fechaFin) => {
        const response = await axios.get(`/dashboard/reportes/asistencia/grado/${idGradoSeccion}?fechaInicio=${fechaInicio}&fechaFin=${fechaFin}`);
        return response.data;
    }
};